tutor/src/components/enroll/index.cjsx

```diff
@@ -0,0 +1,90 @@
+React = require 'react'
+_ = require 'underscore'
+{Promise} = require 'es6-promise'
+
+{CourseEnrollmentActions, CourseEnrollmentStore} = require '../../flux/course-enrollment'
+{CourseActions, CourseStore} = require '../../flux/course'
+ENTER = 'Enter'
+
+ConfirmJoin = require './confirm-join'
+JoinConflict = require './join-conflict'
+Router = require '../../helpers/router'
+
+{MessageList} = require 'shared'
+
+Enroll = React.createClass
+
+  contextTypes:
+    router: React.PropTypes.object
+
+  componentWillMount: ->
+    {enrollmentCode} = Router.currentParams()
+    CourseEnrollmentStore.on('changed', @onCourseEnrollmentChange)
+    CourseEnrollmentActions.create(enrollmentCode)
+
+  componentWillUnmount: ->
+    CourseEnrollmentStore.off('changed', @onCourseEnrollmentChange)
+
```

Comment:
Rather than doing the manual `on`/`off` store dance here, you can use the `bind-store-mixin`, as https://github.com/openstax/tutor-js/blob/f7af0fd8cd0eb17916cb8152aacd28d699c2df20/tutor/src/components/exercises/preview.cjsx#L19-L23 (or search for BindStoreMixin for others).